service/src/main/java/com/ge/predix/acs/service/policy/evaluation/PolicyEvaluationServiceImpl.java

```diff
@@ -97,52 +137,61 @@ public PolicyEvaluationResult evalPolicy(final PolicyEvaluationRequestV1 request
-                            "PolicyEvaluationResult input paramters cannot be null, "
-                                    + "resourceURI=[%s] subjectIdentifier=[%s] action=[%s]",
-                            uri, subjectIdentifier, action));
+    LinkedHashSet<PolicySet> filterPolicySetsByPriority(final String subjectIdentifier, final String uri,
+            final List<PolicySet> allPolicySets, final LinkedHashSet<String> policySetsEvaluationOrder)
+            throws IllegalArgumentException {
 
-            throw new IllegalArgumentException(
-                    "ACS Internal Error: PolicyEvaluationResult input paramters cannot be null.");
+        if (policySetsEvaluationOrder.isEmpty()) {
+            if (allPolicySets.size() > 1) {
+                LOGGER.error(String
+                        .format("Found more than one policy set during policy evaluation and "
+                                + "no evaluation order is provided. subjectIdentifier='%s', resourceURI='%s'",
+                                subjectIdentifier, uri));
+                throw new IllegalArgumentException("More than one policy set exists for this zone. "
+                        + "Please provide an ordered list of policy set names to consider for this evaluation and "
+                        + "resubmit the request.");
+            } else {
+                return allPolicySets.stream().collect(Collectors.toCollection(LinkedHashSet::new));
+            }
         }
 
-        List<PolicySet> allPolicySets = this.policyService.getAllPolicySets();
-
-        if (allPolicySets.isEmpty()) {
-            return new PolicyEvaluationResult(Effect.NOT_APPLICABLE);
-        } else if (allPolicySets.size() > 1) {
-            LOGGER.error("Found more than one policy set during policy evaluation. Subject: " + subjectIdentifier
-                    + ", Resource: " + uri);
-            throw new IllegalArgumentException("More than one policy set exists for this zone. "
-                    + "Remove unnecessary policy sets using DELETE /policy-set/{id} and resubmit request.");
-        } else {
-            // NOTE: When multiple policy sets are supported, this code needs to
-            // delegate to a "InterPolicySetDecisionAggregator"
-            PolicyEvaluationResult result = evalPolicySet(allPolicySets.get(0), subjectIdentifier, uri, action,
-                    supplementalResourceAttributes, supplementalSubjectAttributes);
-
-            LOGGER.info(
-                    String.format(
-                            "Processed Policy Evaluation for: "
-                                    + "resourceUri=[%s], subjectIdentifier=[%s], action=[%s]," + " result=[%s]",
-                            uri, subjectIdentifier, action, result.getEffect()));
-
-            return result;
+        Map<String, PolicySet> allPolicySetsMap = allPolicySets.stream()
+                .collect(Collectors.toMap(PolicySet::getName, Function.identity()));
```

Comment:
@DarioAmiri, last request to convert allPolicySets list to map is implemented.  
All pull requests tests PASSED except cloud-titan profile:
https://predix1.jenkins.build.ge.com/job/Predix-Security/job/acs-build-n-test-pullrequest/676/

Merging branch into develop